src/tests/ftest/util/server_utils.py

```diff
@@ -292,14 +299,21 @@ def run_server(test, hostfile, setname, uri_path=None, env_dict=None,
         # PATH along to the remote
         if build_vars["PREFIX"] != "/usr":
             server_cmd.extend(["-x", "PATH"])
+            default_tmpdir = os.path.join(build_vars["PREFIX"], "tmp")
+        else:
+            default_tmpdir = os.path.expanduser('~/daos_test')
+
+        # build shared dir for tmp amd yaml files
+        tmpdir = os.getenv('DAOS_TEST_SHARED_DIR', default_tmpdir)
 
+        # set env CRT_ATTACH_INFO_PATH
+        os.environ["CRT_ATTACH_INFO_PATH"] = tmpdir
```

Comment:
If we are dynamically assigning the `CRT_ATTACH_INFO_PATH` env, then it needs to be exported by the `orterun` command so the other nodes know it is located.